doc/languages-frameworks/python.md

```diff
@@ -0,0 +1,736 @@
+libraries you would like to use to develop cannot. If you do install libraries
+in your profile, then you will end up with import errors.
+
+#### Python environments using `nix-shell`
+
+The recommended method for creating Python environments for development is with
+`nix-shell`. Executing
+
+```sh
+$ nix-shell -p python35Packages.numpy python35Packages.toolz
+```
+
+opens a Nix shell which has available the requested packages and dependencies.
+Now you can launch the Python interpreter (which is itself a dependency)
+
+```sh
+[nix-shell:~] python3
+```
+
+If the packages were not available yet in the Nix store, Nix would download or
+build them automatically. A convenient option with `nix-shell` is the `--run`
+option, with which you can execute a command in the `nix-shell`. Let's say we
+want the above environment and directly run the Python interpreter
+
+```sh
+$ nix-shell -p python35Packages.numpy python35Packages.toolz --run "python3"
+```
+
+This way you can use the `--run` option also to directly run a script
+
+```sh
+$ nix-shell -p python35Packages.numpy python35Packages.toolz --run "python3 myscript.py"
+```
+
+In fact, for this specific use case there is a more convenient method. You can
+add a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) to your script
+specifying which dependencies Nix shell needs. With the following shebang, you
+can use `nix-shell myscript.py` and it will make available all dependencies and
+run the script in the `python3` shell.
+
+```py
+#! /usr/bin/env nix-shell
+#! nix-shell -i python3 -p python3Packages.numpy
+
+import numpy
+
+print(numpy.__version__)
+```
+
+Likely you do not want to type your dependencies each and every time. What you
+can do is write a simple Nix expression which sets up an environment for you,
+requiring you only to type `nix-shell`. Say we want to have Python 3.5, `numpy`
+and `toolz`, like before, in an environment. With a `shell.nix` file
+containing
+
+```nix
+with import <nixpkgs> {};
+
+(pkgs.python35.buildEnv.override  {
+  extraLibs = with pkgs.python35Packages; [ numpy toolz ];
+}).env
+```
+executing `nix-shell` gives you again a Nix shell from which you can run Python.
+
+What's happening here?
+
+1. We begin with importing the Nix Packages collections. `import <nixpkgs>` import the `<nixpkgs>` function, `{}` calls it and the `with` statement brings all attributes of `nixpkgs` in the local scope. Therefore we can now use `pkgs`.
+2. Then we create a Python 3.5 environment with `pkgs.buildEnv`. Because we want to use it with a custom set of Python packages, we override it.
+3. The `extraLibs` argument of the original `buildEnv` function can be used to specify which packages should be included. We want `numpy` and `toolz`. Again, we use the `with` statement to bring a set of attributes into the local scope.
+4. And finally, for in interactive use we return the environment.
+
+
+#### Declarative environment using myEnvFun
```

Comment:
:+1: i would be also in favor of removing this since it really confuses people. if somebody insists on having this in the manual i would create a separate section about myEnvFun.